### DISCUSSION TITLE
Installation help

### DISCUSSION THREAD

**Discussion by nikitosoz8** [PhlexPlexico/G5API#177] [Help]
Hi!

I wanted to install this on my website, but I was unclear what I am exactly doing. Should I execute those commands on my PC or somehow on my web hosting?

**Comment by PhlexPlexico** [comment]
Heya,

This is all meant for a web server. The API will spin up its own server on a port, and then you would use a front end to interact with it. If your web hosting does not provide you SSH access to that server, then you will not be able to use it. If you're trying to set this all up remotely and assign it an address, I would recommend using a VPS solution like DigitalOcean, Vultr, Linode, etc. 

If the web host only allows cPanel access or similar, I don't believe you can install it on their services. You need SSH access and the ability to install packages on their servers.

Thanks!

**Reply by nikitosoz8** [reply]
My web host has ispmanager which has Shell in it, and it allowed me to run `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash` , but then when I tried to run `nvm install 14.17.5` it said that "nvm" command not found.

**Reply by PhlexPlexico** [reply]
Was there any output from the prior curl command to install nvm? Also, does it allow you to use docker? If so, that may be the easier setup option. If not, then let me know.

**Reply by nikitosoz8** [reply]
Well I just bought a VPS server which allows to use Docker.

**Reply by PhlexPlexico** [reply]
Perfect. You should be able to follow the wiki guide to see an explanation for all the values in the `docker-compose.yml` file. Once they are all filled out (please note, they do not require quotes :)) you can just run `docker compose up -d` and wait a few minutes then you can navigate to the URL to see your web panel and API!

Link to the wiki is [here](https://github.com/PhlexPlexico/G5API/wiki/Configuration#server)
And here is an example docker compose file with what you need to change highlighted by `#CHANGEME` on the same lines [here](https://gist.github.com/PhlexPlexico/e2bbf34507c123ad86a7e6017b441e84)

Thanks!

**Reply by nikitosoz8** [reply]
Do I have to upload Get5 API files somehow for that?

**Reply by PhlexPlexico** [reply]
Technically no :) you can just create a `docker-compose.yml` file on your VPS with that example (filling out the necessary fields) and it should fetch all the latest images from Docker/GHCR I believe (after filling it out and calling `docker-compose up -d`)!
